enumerables.rb

```diff
@@ -0,0 +1,162 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/ModuleLength, Metrics/MethodLength
+module Enumerable
+  def my_each
+    arr = self.class == Range ? Array(self) : self
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < arr.length
+      yield arr[i]
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    arr = self.class == Range ? Array(self) : self
+    return to_enum(:my_each_with_index) unless block_given?
+
+    j = 0
+    while j < arr.length
+      yield arr[j], j
+      j += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    selected = []
+    my_each do |i|
+      result = yield(i)
+      selected << i if result
+    end
+    selected
+  end
+
+  def my_all?(*arg)
+    arr = self.class == Range ? Array(self) : self
+
+    condition = true
+    return true if empty?
+
+    arr.my_each_with_index do |i, j|
+      return true if i && arr[j + 1] && arr[j] == arr[j + 1] && !arg && !block_given?
+
+      case arg[0]
+      when Class
+        condition = false if i.is_a?(arg[0]) == false
+      when Regexp
+        condition = false unless i&.to_s&.match?(arg[0])
+      when Integer || String
+        condition = false if arg[0] != i
+      when Proc
+        condition = false if arg[0].call(i) == false
+      end
+      result = yield(i) if block_given?
+      condition = result if block_given?
+      break if condition == false
+    end
+    condition
+  end
+
+  def my_any?(*arg)
+    arr = self.class == Range ? Array(self) : self
+
+    condition = false
+    return false if empty?
+
+    arr.my_each_with_index do |i, _j|
+      return true if i && !block_given? && !arg[0]
+
+      case arg[0]
+      when Class
+        condition = true if i.is_a?(arg[0])
+      when Regexp
+        condition = true if i&.to_s&.match?(arg[0])
+      when Integer || String
```

Comment:
### Mandatory
**`my_any?`** identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- While I execute `[1, 2, 3].my_any?(1)` It works properly. But as I try to work with sting is does not work properly. for an example->
```ruby
['1', '2', '3'].my_any?('3')    #should return true 
```